sdk/core/Azure.Core/src/Shared/NextLinkOperationImplementation.cs

```diff
@@ -136,7 +138,17 @@ internal class NextLinkOperationImplementation : IOperation
             return new NextLinkOperationImplementation(pipeline, requestMethod, startRequestUri, nextRequestUri, headerSource, lastKnownLocation, finalStateVia, null, rehydrationToken.Id);
         }
 
-        private static string GetContentFromRehydrationToken(Dictionary<string, string> lroDetails, string key)
+        private static IReadOnlyDictionary<string, string?> ConstructLroDetails(BinaryData data)
+        {
+            var lroDetails = new Dictionary<string, string?>();
+            foreach (var property in JsonDocument.Parse(data).RootElement.EnumerateObject())
```

Comment:
The JsonDocument will need to be disposed.

Reply:
Right, updated.